lib/puppet/provider/network_config/interfaces.rb

```diff
@@ -29,24 +29,22 @@ def self.target_files
 
   class MalformedInterfacesError < Puppet::Error
     def initialize(msg = nil)
-      msg = 'Malformed debian interfaces file; cannot instantiate network_config resources' if msg.nil?
+      msg = 'Malformed debian interfaces file; cannot instantiate network_config resources' unless msg
```

Comment:
I'm not a ruby guru but wouldn't msg ||= 'Malformed debian interfaces file; cannot instantiate network_config resources'
work here?